problem_builder/mentoring.py

```diff
@@ -156,18 +161,28 @@ def get_content_titles(self):
             return [self.display_name]
         return []
 
-    def get_theme(self):
+    def get_settings(self, settings_key, default):
         """
-        Gets theme settings from settings service. Falls back to default (LMS) theme
-        if settings service is not available, xblock theme settings are not set or does
-        contain mentoring theme settings.
+        Get settings identified by `settings_key` from settings service.
+
+        Fall back on `default` if settings service is unavailable
+        or settings have not been customized.
         """
         settings_service = self.runtime.service(self, "settings")
         if settings_service:
             xblock_settings = settings_service.get_settings_bucket(self)
-            if xblock_settings and self.theme_key in xblock_settings:
-                return xblock_settings[self.theme_key]
-        return _default_theme_config
+            if xblock_settings and settings_key in xblock_settings:
+                return xblock_settings[settings_key]
+        return default
+
+    def get_theme(self):
```

Comment:
Likewise for get_theme here and all. I'm kind of surprised this isn't already done. Am I missing something or did we just make these mixins after the fact?